### PR TITLE
Also verify signature when adding to transaction

### PR DESCRIPTION
**Pull request by TorstenStueber**
This PR will add code to verify a signature when added to a transaction.

This is valuable for scenarios where a party receives a signature from another party and wants to add this signature to a transaction. The first party needs to make sure that the signature from the other party is valid.

This will become apparent once the transaction is submitted to horizon – it will fail if the signature is invalid. However, for smart contract schemes (say, payment channels) this will usually happen much later.

### CODE REVIEW

File: src/transaction.js [hunk 1]
```diff
@@ -130,18 +131,25 @@ export class Transaction {
...
 
     try {
-      hint = Keypair.fromPublicKey(publicKey).signatureHint();
+      keypair = Keypair.fromPublicKey(publicKey);
+      hint = keypair.signatureHint();
     } catch (e) {
       throw new Error('Invalid publicKey');
     }
 
+    if (!keypair.verify(this.hash(), signatureBuffer)) {
```
Comment (vcarl) [inline src/transaction.js]: This isn't a safe assumption to make. This only validates the "master key", but doesn't account for other signers that may be on the account. See the [docs on multisignature](https://www.stellar.org/developers/guides/concepts/multi-sig.html#additional-signing-keys) for more info.

Comment (TorstenStueber) [reply src/transaction.js]: I am not sure I understand your comment. This verifies that the provided `signature` is a valid signature of the keypair provided by the `publicKey` for this transaction.

So this does not verify the master key but it verifies the signer that provided the signature – as intended. Am I missing something?

**Comment by TorstenStueber** [comment]
@tomerweller This is a valid concern. But even though adding a signature to a transaction is not technically depending on the network, it is implicitly doing so. A combination of a signature and a transaction only makes sense for a specific network, otherwise the signature would be invalid. I don't see a use case where one wants to add a signature to a transaction while not having a specific network in mind (namely the one where the signature is valid).

**Comment by TorstenStueber** [comment]
@bartekn @morleyzhi any comments from you on this PR? Who would be responsible to close or accept this PR?

**Pull request merged**